### DISCUSSION TITLE
关于有些异步 API 的问题

### DISCUSSION THREAD

**Discussion by Arondight** [takayama-lily/oicq#278] [Ideas]
抱歉又打扰你了，有的异步我觉得没有任何意义，例如 `await client.getGroupInfo(msg.group_id)` 和 `client.pickGroup(msg.group_id).info || {}`，强行异步只会增加使用成本，设计 API 的时候或许更多的考虑了调用形式的一致？我觉得这些 API 有一个 Sync 版本会更好一些，个人觉得除非必要尽量不异步，另外我确实没搞明白为啥以前这是个同步的 API 现在变异步了，因为我是在同步环境中用的所以只能去使用本来不应该暴露出来的方法，这样有点奇怪。

**Comment by takayama-lily** [comment]
`getGroupInfo`这个方法一直都是异步的，因为可能需要更新，或者重新获取

**Reply by Arondight** [reply]
谢谢，原来如此，不过1.x这个确实是同步的

**Reply by takayama-lily** [reply]
你是不是弄错了，刚才去看了一眼，返回的都是`Promise`

**Reply by Arondight** [reply]
看了下是我搞错了，1.x 我用这个的时候还没学过js，我以为不带async就是同步，大概我一直用的promise做比较的

**Reply by Cloud370** [reply]
哈哈哈哈哈 几年前刚学Node.js的时候我也是这么想的 但是node涉及到网络等耗时操作就全都是Promise了 只能说习惯就好 现在写一个function 不管是不是promise我都习惯先写async

**Reply by wuxianggujun** [reply]
node.js不都是异步的吗？♥(｡￫v￩｡)♥
